authentication/src/main/java/io/camunda/authentication/oauth/PersistedOAuth2AuthorizedClientService.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.oauth;
+
+import io.camunda.search.clients.PersistentOAuth2AuthorizedClientsClient;
+import io.camunda.search.entities.PersistentOAuth2AuthorizedClientEntity;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken;
+
+public class PersistedOAuth2AuthorizedClientService implements OAuth2AuthorizedClientService {
```

Comment:
How do we do housekeeping of the persisted clients? For example, if users abandon their sessions, how will entries get removed again from the secondary storage? For sessions we have `io.camunda.authentication.session.WebSessionDeletionTask`.

Reply:
We don't. I was thinking that we don't need it. The reason being session is much more transient entity, while client is something that stays for a long time. Also there's not that much of clients, as sessions.

To summarize my point, cleaning up sessions will cause too big complication for very small gain.

Reply:
For transparency, via private chat, Igor and I agreed to have some housekeeping logic. Reason being that it's general best practice that any data the system creates can get deleted again (either through API or internal behavior). Even if this will very likely not have a high volume of data, it's a risk to leave this entirely uncontrolled.